### PR TITLE
EMPLOYMENT_WA_1_0 Release/feature branch

### PR DESCRIPTION
**Pull request by mikebrownccd**
### JIRA link (if applicable) ###

* [AM-2688](https://tools.hmcts.net/jira/browse/AM-2688) _"ET_WA - Add mapping rules in ORM for STAFF users"_
   * [AM-2615](https://tools.hmcts.net/jira/browse/AM-2615) _"ET_WA - Drool JUnit testing in ORM and RAS applications for STAFF users"_ 
* [AM-2612](https://tools.hmcts.net/jira/browse/AM-2612) _"ET_WA - Add mapping rules in ORM for ET judges"_
   * [AM-2687](https://tools.hmcts.net/jira/browse/AM-2687) _"ET_WA - Drool JUnit testing in ORM and RAS applications for JUDICIAL users"_
* [AM-2831](https://tools.hmcts.net/jira/browse/AM-2831) _"ET_WA - Final round of review and testing of combined feature branches for release Employment_WA_1.0"_ 

### Change description ###

EMPLOYMENT_WA_1_0 Release/feature branch:

* #1410
* #1407
* #1470

> NB: This PR is has a dependency on hmcts/am-role-assignment-service#1863

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/validationrules/core/hearing-role-judicial-global.drl [hunk 1]
```diff
@@ -26,20 +26,8 @@ import function uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMapping
...
-   "PUBLICLAW High Court Judge - Sitting in Retirement - Fee Paid",
-   "PUBLICLAW Magistrate - Voluntary", "PUBLICLAW Designated Family Judge - Salaried",
-   "PUBLICLAW Deputy Circuit Judge - Fee Paid", "PUBLICLAW District Judge (MC) - Salaried", "PUBLICLAW Recorder - Fee Paid",
-   "CIVIL Circuit Judge-Salaried","CIVIL Deputy Circuit Judge-Fee-Paid","CIVIL Specialist Circuit Judge-Salaried",
-   "CIVIL Deputy District Judge-Fee-Paid","CIVIL Deputy District Judge - Sitting in Retirement-Fee-Paid",
-   "CIVIL District Judge-Salaried","CIVIL Recorder-Fee-Paid","CIVIL Senior Circuit Judge-Salaried",
-   "CIVIL High Court Judge-Salaried","CIVIL Designated Civil Judge-Salaried","CIVIL Presiding Judge-Salaried",
-   "CIVIL Resident Judge-Salaried"))
+   $joh: JudicialOfficeHolder(office in ("SSCS Judge", "Family Judge") ||
+        jurisdiction in ("PUBLICLAW", "CIVIL", "EMPLOYMENT"))
```
Comment (mattnayler) [inline src/main/resources/validationrules/core/hearing-role-judicial-global.drl:30]: NB: Current Confluence for "hearing-viewer" roles for ET say: "Any Employment JOH", for Civil say: "Any valid Judicial Appointment/Role", for Family Public Law say :"Any Judicial Appointment".

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolEmploymentJudicialRoleMappingTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    @ParameterizedTest
+    @CsvSource({
+            "EMPLOYMENT President of Tribunal-Salaried,leadership-judge,judge,task-supervisor,case-allocator,"
+                    + "hmcts-judiciary,specific-access-approver-judiciary",
+            "EMPLOYMENT Vice President-Salaried,leadership-judge,judge,task-supervisor,case-allocator,"
+                    + "hmcts-judiciary,specific-access-approver-judiciary",
+            "EMPLOYMENT Regional Employment Judge-Salaried,leadership-judge,judge,task-supervisor,case-allocator,"
+                    + "hmcts-judiciary,specific-access-approver-judiciary"
+    })
+    void shouldReturnPresidentOfTribunalVicePresidentRegionalEmploymentJudgeSalariedRolesRoles(String setOffice,
```
Comment (mattnayler) [inline src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolEmploymentJudicialRoleMappingTest.java:104]: Should this be renamed as it ends with `RolesRoles`?

File: src/main/resources/validationrules/employment/employment-admin-mapping.drl [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+end;
+
+/*
+ * employment admin "hmcts-admin" Org role mapping.
+ */
+
+rule "employment_admin_hmcts_admin_org_role"
+when
+  $f:  FeatureFlag(status && flagName == FeatureFlagEnum.EMPLOYMENT_WA_1_0.getValue())
+  $cap: CaseWorkerAccessProfile(roleId in ("3","12"), serviceCode == "BHA1", !suspended)
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/validationrules/employment/employment-admin-mapping.drl:142]: Resolved in #1470 see 5dcdc0d3f5916d5637b726fd22a4f55999d74d60.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolEmploymentStaffOrgRolesTest.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@RunWith(MockitoJUnitRunner.class)
+class DroolEmploymentStaffOrgRolesTest extends DroolBase {
+
+    @ParameterizedTest
+    @CsvSource({
+            "3,BHA1,'hearing-centre-team-leader,hearing-centre-admin,hmcts-admin,specific-access-approver-admin',N,N",
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolEmploymentStaffOrgRolesTest.java:29]: Resolved in #1470 see 5dcdc0d3f5916d5637b726fd22a4f55999d74d60.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolEmploymentStaffOrgRolesTest.java [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+                    if (roleNamesWithRegionAttribute.contains(r.getRoleName())) {
+                        assertEquals("LDN", r.getAttributes().get("region").asText());
+                    }
+                    //assert work types
+                    if (("hearing-centre-team-leader").equals(r.getRoleName())) {
+                        assertNull(r.getAttributes().get("workTypes"));
+                    } else if (("hearing-centre-admin").equals(r.getRoleName())) {
+                        assertEquals("hearing_work,routine_work,applications,amendments",
+                                r.getAttributes().get("workTypes").asText());
+                    }
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolEmploymentStaffOrgRolesTest.java:87]: Resolved in #1470 see 5dcdc0d3f5916d5637b726fd22a4f55999d74d60.

File: src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolEmploymentStaffOrgRolesTest.java [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+        roleAssignments.stream().filter(c -> c.getGrantType().equals(GrantType.STANDARD)).toList()
+                .forEach(r -> {
+                    assertEquals("EMPLOYMENT", r.getAttributes().get("jurisdiction").asText());
+                    assertEquals(cap.getPrimaryLocationId(), r.getAttributes().get("primaryLocation").asText());
+                    //assert work types
+                    if (("ctsc").equals(r.getRoleName())) {
+                        assertEquals("hearing_work,routine_work,applications,amendments",
+                                r.getAttributes().get("workTypes").asText());
+                    }
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolEmploymentStaffOrgRolesTest.java:146]: Resolved in #1470 see 5dcdc0d3f5916d5637b726fd22a4f55999d74d60.

**Review by tomxelliott — APPROVED** [review]
LGTM, cross checked with config on Confluence

**Comment by mattnayler** [comment]
Updated with `master` in 68bf88bb7375b85fffb3abd2bdde05aa981b974a: no conflicts or changes required.

**Pull request merged**